src/perfEvents_linux.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PERFEVENTS_LINUX_HPP
```

Comment:
I assume this was moved to a separate file to make it possible to call PerfEventType functions from a test, right?
I'm afraid this approach does not scale well. Ideally, unit tests should be completely non-intrusive, i.e., do not require refactoring. Is it possible to avoid exposing implementation details through interfaces solely for the purpose of testing? Perhaps, by merging sources.

Reply:
Don't build fat binary for unit tests then? Alternatively, merge sources to a temporary file rather than pipe or use some other build-time trick.

Extracting headers is undesirable, because it's more code, additional manual work, and exhibiting of implementation details that could otherwise stay local to the place of use.